src/main/kotlin/io/newm/kogmios/ClientImpl.kt

```diff
@@ -179,199 +181,232 @@ internal class ClientImpl(
-                                            requestNextRequestResponseMap.remove(response.reflection)
-                                                ?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
-                                        }
-
-                                        is MsgSubmitTxResponse -> {
-                                            submitTxResponseMap.remove(response.reflection)?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
-                                        }
-
-                                        is MsgEvaluateTxResponse -> {
-                                            evaluateTxResponseMap.remove(response.reflection)?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
-                                        }
-
-                                        is MsgAwaitAcquireMempoolResponse -> {
-                                            awaitAcquireResponseMap.remove(response.reflection)?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
-                                        }
-
-                                        is MsgReleaseMempoolResponse -> {
-                                            releaseMempoolResponseMap.remove(response.reflection)?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
-                                        }
-
-                                        is MsgHasTxResponse -> {
-                                            hasTxResponseMap.remove(response.reflection)?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
-                                        }
-
-                                        is MsgSizeAndCapacityResponse -> {
-                                            sizeAndCapacityResponseMap.remove(response.reflection)?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
-                                        }
-
-                                        is MsgNextTxResponse -> {
-                                            nextTxResponseMap.remove(response.reflection)?.complete(response)
-                                                ?: log.warn("No handler found for: ${response.reflection}")
+                    delay(1000) // wait a bit in case server disconnects us immediately.
+                    if (!coroutineContext.job.isActive) {
+                        result.complete(false)
+                    } else {
+                        session = this
+                        _isConnected = true
```

Comment:
Unrelated to this fix, but one way to get rid of having both `_isConnected` and `isConnected` is:
```
    override var isIt: Boolean = false
        private set

```